src/App.tsx

```diff
@@ -14,11 +14,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  setSelectedGoods = (good: string) => {
+    const { selectedGoods } = this.state;
+
+    this.setState(((prevState: State) => {
+      if (selectedGoods.includes(good)) {
+        return {
+          selectedGoods: prevState.selectedGoods
+            .filter(prevGood => prevGood !== good),
+        };
+      }
+
+      return { selectedGoods: [...prevState.selectedGoods, good] };
+    }));
+  };
+
+  clearSelectedGood = (good: string) => {
+    this.setState((prevState => {
+      return {
+        selectedGoods: [...prevState.selectedGoods].filter(e => e !== good),
+      };
+    }));
```

Comment:
why not implicit return? I found out today that it's a big deal :)

```suggestion
    this.setState((prevState => ({
        selectedGoods: [...prevState.selectedGoods].filter(e => e !== good),
      };
    )));
```